akka-projection-core/src/main/scala/akka/projection/Projection.scala

```diff
@@ -11,7 +11,35 @@ import akka.annotation.ApiMayChange
 import scala.concurrent.{ ExecutionContext, Future }
 
 @ApiMayChange
-trait Projection {
+trait Projection[StreamElement] {
+
+  /**
+   * The projection group name. The name must be unique over the whole system.
+   * For instance, a 'user-view' could be the name of a projection.
+   */
+  def groupName: String
+
+  /**
+   * The unique key for this Projection instance. Must be unique within the projection group.
+   */
+  def key: String
+
+  /**
+   * The method wrapping the user EventHandler function.
+   */
+  def processElement(elt: StreamElement): Future[StreamElement]
+
+  /**
+   * Start to run a Projection
+   *
+   * @param systemProvider
+   */
   def start()(implicit systemProvider: ClassicActorSystemProvider): Unit
```

Comment:
An alternative would be to have the system as a `def systemProvider: ClassicActorSystemProvider`. Might be easier if it's needed in `stop` implementation.

Reply:
Yep, let's keep this in mind. We can change on another PR.